builtin/aws/ecs/platform.go

```diff
@@ -1301,14 +1308,48 @@ func (p *Platform) resourceTargetGroupCreate(
 
 	state.Port = p.config.ServicePort
 
-	ctg, err := elbsrv.CreateTargetGroupWithContext(ctx, &elbv2.CreateTargetGroupInput{
+	if p.config.HealthCheck.Protocol == "" {
+		p.config.HealthCheck.Protocol = "HTTP"
+	}
+
+	createTargetGroupInput := &elbv2.CreateTargetGroupInput{
 		HealthCheckEnabled: aws.Bool(true),
 		Name:               &targetGroupName,
 		Port:               &state.Port,
-		Protocol:           aws.String("HTTP"),
+		Protocol:           aws.String(p.config.HealthCheck.Protocol),
 		TargetType:         aws.String("ip"),
 		VpcId:              &subnets.Subnets.VpcId,
-	})
+	}
+
+	if p.config.HealthCheck.Path != "" {
+		createTargetGroupInput.HealthCheckPath = aws.String(p.config.HealthCheck.Path)
+	}
+
+	if p.config.HealthCheck.Timeout != 0 {
+		createTargetGroupInput.HealthCheckTimeoutSeconds = aws.Int64(p.config.HealthCheck.Timeout)
+	}
+
+	if p.config.HealthCheck.Interval != 0 {
+		createTargetGroupInput.HealthCheckIntervalSeconds = aws.Int64(p.config.HealthCheck.Interval)
+	}
+
+	if p.config.HealthCheck.HealthyThresholdCount != 0 {
+		createTargetGroupInput.HealthyThresholdCount = aws.Int64(p.config.HealthCheck.HealthyThresholdCount)
```

Comment:
Where is this default set? If it's the default that AWS gives if we do not supply a value, then we'll need to set it ourselves here `HealthyThresholdCount == 0`. Otherwise if a user decides to remove their custom value, they might assume that it will revert to `5` because that's what our docs say, but it wont, because it will be `0` in our config and then this block wont execute.

Reply:
This is indeed the default from AWS, and what you're saying definitely makes sense. I updated this to explicitly set the value to 5 if the user doesn't supply a value (the `HealthyThresholdCount` ends up being 0).